ggml-cuda.cu

```diff
@@ -6258,6 +6265,41 @@ static void ggml_cuda_op_flatten(const ggml_tensor * src0, const ggml_tensor * s
     }
 }
 
+void ggml_cuda_set_peer_access(const int n_tokens) {
+    static int last_n_tokens = INT_MAX;
+
+    if (n_tokens == last_n_tokens) {
+        return;
+    }
+
+#ifdef NDEBUG
```

Comment:
With the current code base peer access is only enabled in order to allow direct GPU<->GPU data transfer without going through the CPU. We explicitly do **not** want to use the feature where data from another GPU can be accessed directly because in my testing that had bad performance. So we would want a way to detect these memory accesses. If peer access is disabled for debug builds where performance is irrelevant anyways accidentally accessing data from another device will cause an error and can thus be detected.